### PR TITLE
(rcm-88) receive and send back a "client_state" controlled by the backend

### PR DESCRIPTION
**Pull request by arbll**
…kend

### What does this PR do?

This PR reads a `client_state` field in the custom of the director targets and sends it back to the backend if present.

### Motivation

This will allow the backend to ask the client to store state data about itself in between poll queries. This allows us as an example to track the list of files the agent already has. The goal with this arbitrary field instead of just sending a list of file from the agent for example is that we can change what that data contains anytime and be forward compatible with any change in how the backend distributes configs.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by mellon85 — APPROVED** [review]
LGTM

File: pkg/proto/datadog/remoteconfig/remoteconfig.proto [hunk 1]
```diff
@@ -49,6 +49,7 @@ message LatestConfigsRequest {
   repeated string products = 4;
   repeated string new_products = 5;
   repeated Client active_clients = 6;
+  bytes client_state = 10;
```
Comment (ameske) [inline pkg/proto/datadog/remoteconfig/remoteconfig.proto:52]: Why 10, as opposed to just ++'ing to 7?

**Review by ameske — APPROVED** [review]

File: pkg/proto/datadog/remoteconfig/remoteconfig.proto (hunk 1 above)
Comment (arbll) [reply pkg/proto/datadog/remoteconfig/remoteconfig.proto:52]: Fields are not in order unfortunately

Comment (ameske) [reply pkg/proto/datadog/remoteconfig/remoteconfig.proto:52]: Ahhh, minimal diff surrounding context fail on my part. 😁 Thanks!

**Pull request merged**